src/TemplateRenderer.php

```diff
@@ -18,32 +18,29 @@
 use function str_replace;
 
 /**
- * `ViewRenderer` allows using Twig with a View service.
+ * TemplateRenderer allows using Twig with a View service.
  */
-final class ViewRenderer implements TemplateRendererInterface
+final class TemplateRenderer implements TemplateRendererInterface
 {
-    public function __construct(private Environment $environment)
+    public function __construct(private readonly Environment $environment)
     {
     }
 
     public function render(ViewInterface $view, string $template, array $parameters): string
     {
-        $environment = $this->environment;
-        $renderer = function () use ($view, $template, $parameters, $environment): void {
-            $template = str_replace('\\', '/', $template);
-            $basePath = str_replace('\\', '/', $view->getBasePath());
-            $file = str_replace($basePath, '', $template);
-
-            echo $environment->render($file, array_merge($parameters, ['this' => $view]));
-        };
+        $templateFile = str_replace(
```

Comment:
What if `$template` placed not in base path?

Reply:
Throws an exception if base path of `$template` doesn't exists in twig loader.